### PR TITLE
Add authentication via query string

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Implement authentication via query string, i.e., like
```
curl http://pi.hole/api/info/version?sid=%2FodNaBgsMlSwAspmw25t0g%3D
```
The `sid` should be URL-encoded for most applications (like `python3 requests` or `curl`).

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Review by PromoFaux — APPROVED** [review]
Looks straightforward enough

File: src/api/auth.c [hunk 1]
```diff
@@ -151,7 +152,22 @@ int check_client_auth(struct ftl_conn *api, const bool is_api)
 			// Mark SID as available
 			sid_avail = true;
 		}
+	}
 
+	// If not, does the client provide a session ID via URI?
+	if(!sid_avail && api->request->query_string && GET_VAR("sid", sid, api->request->query_string) > 0)
+	{
+		// "+" may have been replaced by " ", undo this here
```
Comment (yubiuser) [inline src/api/auth.c:160]: Where should such a conversion happen? (`+` -> ` `)

Comment (DL6ER) [reply src/api/auth.c:160]: https://github.com/pi-hole/FTL/blob/a5d0a663e40ed12d0d5ee608ff2523cd5208a908/src/civetweb/civetweb.c#L14096-L14101

We typically want decoding to happen, so we cannot simply disable this

**Review by yubiuser — APPROVED** [review]
Works as expected. First try is with a different branch.

```
chrko@ThinkPad-X230:~$ curl -s http://nanopi.lan/api/info/version?sid=Mv1bXmCZS4jFzjduaGytFA%3D | jq .
{
  "error": {
    "key": "unauthorized",
    "message": "Unauthorized",
    "hint": null
  },
  "took": 0.001401662826538086
}
chrko@ThinkPad-X230:~$ curl -s http://nanopi.lan/api/info/version?sid=Mv1bXmCZS4jFzjduaGytFA%3D | jq .
{
  "version": {
    "core": {
....

```

**Pull request merged**